src/target.rs

```diff
@@ -30,6 +33,12 @@ pub trait Target: Clone + Copy + std::hash::Hash + Eq + Debug + 'static {
 
     /// Yield target-specific actions which apply to a given [LogicalSpec].
     fn actions(spec: &LogicalSpec<Self>) -> Box<dyn Iterator<Item = Action<Self>>>;
+
+    /// Get corresponding [Targets] enum
+    fn by_enum() -> Targets;
```

Comment:
Rather than doing this manually, can we use `enum_dispatch` like we do elsewhere?

Reply:
If I understand `enum_dispath` correctly, it automatically implements the `From` trait for the enum. What we want to do here is to get the corresponding enum without actual object storage (`From` requires it). This `by_enum` can return from the Target trait or types that implement the Target trait. For example, in `codegen/cpu.rs:36`, we did not need to path the actual object of the Target trait.

Reply:
The object storage is not a problem since `Target`s are zero size; one can just `T::default().into()`. But I see no there is a bigger reason this doesn't work: `Target` has an associated type, so there can't be a same trait impl for the enum `Targets`.

Instead, let's stick with your approach, but give them clearer names:
- [ ] `by_enum` -> `target_id`
- [ ] `Targets` -> `TargetId`

If this gets unwieldy or hard-to-extend in the future, we can switch to using an approach like [inventory](https://docs.rs/inventory/latest/inventory/), but this is good for now.